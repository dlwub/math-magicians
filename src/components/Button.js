import React from 'react';
import '../index.css';

export default class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { value, className } = this.props;
    return (
      <button type="button" className={className}>
        {value}
      </button>
    );
  }
}
