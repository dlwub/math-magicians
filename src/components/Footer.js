import React from 'react';

const Footer = () => (
  <div className="footer">&copy;Desta Legesse Wubishet</div>
);

export default Footer;
