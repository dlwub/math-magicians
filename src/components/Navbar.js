import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <>
    <div className="nav-container">
      <h1>Math Magicians</h1>
      <ul className="nav-list">
        <li><Link to="/">Home </Link></li>
        <li className="calculator-link"><Link to="/calculator">Calculator </Link></li>
        <li><Link to="/quote">Quote </Link></li>
      </ul>
    </div>
  </>
);

export default Navbar;
