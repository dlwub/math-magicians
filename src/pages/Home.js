import React from 'react';

const Home = () => (
  <div className="home-page">
    <p className="welcome-message">Welcome to our page!</p>
    <p className="home-paragraph">
      Math Magicians is a website where Math fans can read famous
      quotes on Mathematics and use a calculator to make computations.
    </p>
    <div className="img-container" />
  </div>
);

export default Home;
