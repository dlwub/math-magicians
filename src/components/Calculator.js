import React from 'react';
import Button from './Button';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="button-container">
        <Button value={0} className="btn row1" />
        <Button value="A/C" className="btn" />
        <Button value="+/-" className="btn" />
        <Button value="%" className="btn" />
        <Button value="รท" className="btn orange" />
        <Button value={7} className="btn" />
        <Button value={8} className="btn" />
        <Button value={9} className="btn" />
        <Button value="ร" className="btn orange" />
        <Button value={4} className="btn" />
        <Button value={5} className="btn" />
        <Button value={6} className="btn" />
        <Button value="-" className="btn orange" />
        <Button value={1} className="btn" />
        <Button value={2} className="btn" />
        <Button value={3} className="btn" />
        <Button value="+" className="btn orange" />
        <Button value={0} className="btn col2" />
        <Button value="." className="btn" />
        <Button value="=" className="btn orange" />

      </div>
    );
  }
}
