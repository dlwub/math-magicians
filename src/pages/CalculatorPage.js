import React from 'react';
import Calculator from '../components/Calculator';

const CalculatorPage = () => (
  <div className="calculator-container">
    <p className="calculator-motto">Let&#39;s do some Math!</p>
    <Calculator className="calculator" />
  </div>
);

export default CalculatorPage;
