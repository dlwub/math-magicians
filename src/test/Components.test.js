import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Button from '../components/Button';
import Calculator from '../components/Calculator';
import Footer from '../components/Footer';
import Navbar from '../components/Navbar';
import Home from '../pages/Home';
import Quote from '../pages/Quote';
import CalculatorPage from '../pages/CalculatorPage';

describe('Components', () => {
  it('Button should match snapshot when empty', () => {
    const button = renderer
      .create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Button should match snapshot when values passed in', () => {
    const value = '7';
    const className = 'btn';
    const button = renderer
      .create(<Button value={value} className={className} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Calculator should match snapshot', () => {
    const calculator = renderer
      .create(<Calculator />).toJSON();
    expect(calculator).toMatchSnapshot();
  });

  it('Navbar should match snapshot', () => {
    <Router>
      const navbar = renderer
      .create(
      <Navbar />
      ).toJSON();
      expect(navbar).toMatchSnapshot();
    </Router>;
  });

  it('Home snapshot should match', () => {
    const home = renderer.create(<Home />).toJSON();
    expect(home).toMatchSnapshot();
  });

  it('Quote snapshot should match', () => {
    const quote = renderer.create(<Quote />).toJSON();
    expect(quote).toMatchSnapshot();
  });

  it('CalculatorPage snapshot should match', () => {
    const calculatorpage = renderer.create(<CalculatorPage />).toJSON();
    expect(calculatorpage).toMatchSnapshot();
  });

  it('Footer snapshot should match', () => {
    const footer = renderer.create(<Footer />).toJSON();
    expect(footer).toMatchSnapshot();
  });
});
