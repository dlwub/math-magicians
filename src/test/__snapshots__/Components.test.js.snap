// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Button should match snapshot when empty 1`] = `
<button
  type="button"
/>
`;

exports[`Components Button should match snapshot when values passed in 1`] = `
<button
  className="btn"
  type="button"
>
  7
</button>
`;

exports[`Components Calculator should match snapshot 1`] = `
<div
  className="button-container"
>
  <button
    className="btn row1"
    type="button"
  >
    0
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="btn orange"
    onClick={[Function]}
    type="button"
  >
    ÷
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="btn orange"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="btn orange"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="btn orange"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="btn col2"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="btn orange"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`Components CalculatorPage snapshot should match 1`] = `
<div
  className="calculator-container"
>
  <p
    className="calculator-motto"
  >
    Let's do some Math!
  </p>
  <div
    className="button-container"
  >
    <button
      className="btn row1"
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn col2"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Components Footer snapshot should match 1`] = `
<div
  className="footer"
>
  ©Desta Legesse Wubishet
</div>
`;

exports[`Components Home snapshot should match 1`] = `
<div
  className="home-page"
>
  <p
    className="welcome-message"
  >
    Welcome to our page!
  </p>
  <p
    className="home-paragraph"
  >
    Math Magicians is a website where Math fans can read famous quotes on Mathematics and use a calculator to make computations.
  </p>
  <div
    className="img-container"
  />
</div>
`;

exports[`Components Quote snapshot should match 1`] = `
<div
  className="quote-container"
>
  <p
    className="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston.
  </p>
</div>
`;
